daffodil-core/src/main/scala/org/apache/daffodil/core/dsom/DFDLSchemaFile.scala

```diff
@@ -206,23 +207,9 @@ final class DFDLSchemaFile(
   private lazy val loader = new DaffodilXMLLoader(errHandler)
 
   lazy val iiXMLSchemaDocument = LV('iiXMLSchemaDocument) {
-    val res = makeXMLSchemaDocument(seenBefore, Some(this))
-    if (res.isDFDLSchema && sset.shouldValidateDFDLSchemas) {
-      //
-      // We validate DFDL schemas, only if validation is requested.
-      // Some things, tests generally, want to turn this validation off.
-      //
-      try {
-        loader.validateAsDFDLSchema(schemaSource)
```

Comment:
This is the big improvement. There is no need to validate each schema file as a DFDL schema, only the root file. This was being called 995 times for Link16.